File: leetcode/1380.Lucky Numbers in a Matrix/solution.rs
impl Solution {
    pub fn lucky_numbers (matrix: Vec<Vec<i32>>) -> Vec<i32> {
        let mut max_in_col : Vec<i32> = vec![i32::MIN; matrix[0].len()];
        let mut min_in_row : Vec<i32> = vec![i32::MAX; matrix.len()];
        for i in 0..matrix.len() {
            for j in 0..matrix[i].len() {
                min_in_row[i] = min_in_row[i].min(matrix[i][j]);
                max_in_col[j] = max_in_col[j].max(matrix[i][j]);
            }
        }
        let mut lucky_numbers : Vec<i32> = Vec::new();
        for i in 0..matrix.len() {
            for j in 0..matrix[i].len() {
                if matrix[i][j] == min_in_row[i] && matrix[i][j] == max_in_col[j] {
                    lucky_numbers.push(matrix[i][j]);
                }
            }
        }
        lucky_numbers
    }
}

File: leetcode/1380.Lucky Numbers in a Matrix/solution.cpp
using namespace std;

class Solution {
public:
    vector<int> luckyNumbers (vector<vector<int>>& matrix) {
        int m = matrix.size();
        int n = matrix[0].size();
        vector<int> row_min(m, INT_MAX);
        vector<int> col_max(n, INT_MIN);
        vector<int> lucky_numbers;
        
        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                row_min[i] = min(row_min[i], matrix[i][j]);
                col_max[j] = max(col_max[j], matrix[i][j]);
            }
        }
        
        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                if (matrix[i][j] == row_min[i] && matrix[i][j] == col_max[j]){
                    lucky_numbers.push_back(matrix[i][j]);
                }
            }
        }
        
        return lucky_numbers;
    }
};